### PR TITLE
Add author

### LINKED ISSUE #5: Add author
Notes:
- Using this as reference: https://simpleisbetterthancomplex.com/tutorial/2016/07/22/how-to-extend-django-user-model.html

### LINKED ISSUE #10: As a server admin, I want to be able to add, modify, and remove authors

### PR DESCRIPTION
**Pull request by TurnipXenon**
Closes #5 
Closes #10

## Details
- Base author object on https://github.com/abramhindle/CMPUT404-project-socialdistribution/blob/master/project.org#authors
- Author app
- Implement URL: ://service/authors/ 

## How to test
1. Install pillow in your virtualenv -> `pip install Pillow`
2. Install SQLite driver (we can change our database later)
3. Do migrations
```bash
python manage.py migrate
```
4. Create superuser
```bash
python manage.py createsuperuser
```
5. Run server
```bash
python manage.py runserver
```
6. Open the url in your browser: `localhost:8080/admin` and login with your credentials. Add, modify, delete authors as you want.

## TODO
- [x] Document how to test this
  - [x] Migrate on their end
  - [x] Create super user
  - [x] Get to the admin panel 
- [x] Add fields for author
- [x] For fellow Pycharm users, document how to fix setting up the project: https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000596250-ImportError-in-Django-App-only-when-I-start-server-from-IDE
- [x] Add requirements.txt

### Admin panel
mysocial_user

### Get all authors
image

## References or what I had to do to do stuff
- https://stackoverflow.com/questions/46755679/django-templatedoesnotexist-error

### CODE REVIEW

**Comment by amandanguyeen** [comment]
okay just to clarify: Server admin is not something we're coding in like an author like it literally means us as developers can use the django admin panel to add, delete and modify users?

File: mysocial/authors/serializers/author_serializer.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    """
+    type = serializers.SerializerMethodField('get_type')
+    displayName = serializers.CharField(source='display_name')
+
+    def get_type(self, model):
+        return model.get_serializer_field_name()
+
+    class Meta:
+        model = Author
+        # todo(turnip): url, host, profile image
```
Comment (amandanguyeen) [inline mysocial/authors/serializers/author_serializer.py:19]: would it be easier to include url, host and profile image now but make it nullable?

**Comment by amandanguyeen** [comment]
I believe the url path should be service/authors.

File: mysocial/authors/views.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        serializer = AuthorSerializer(authors, many=True)
+        # todo(turnip): there might be a better way doing this that can make it surface in the auto docs via serializer
+        return Response({
+            'type': 'authors',
+            'items': serializer.data
+        })
+        # formatted_list = []
+        # for el in Author.objects.all():
+        #     formatted_list.append(el.as_json())
+        # data = {"data": formatted_list}
```
[the remark this answers is not in the document]
Comment (TurnipXenon) [reply mysocial/authors/views.py:27]: ahh did not see that lol

**Comment by TurnipXenon** [comment]
> I believe the url path should be service/authors.

Ahhh... I assumed `service` was a placeholder for our domain. But I could easily add that.

**Comment by TurnipXenon** [comment]
> okay just to clarify: Server admin is not something we're coding in like an author like it literally means us as developers can use the django admin panel to add, delete and modify users?

@amandanguyeen that's what i also assumed

**Comment by amandanguyeen** [comment]
It looks like there's actually two parameters for GET, size and page

**Comment by TurnipXenon** [comment]
Okay, I added size and page now :D

**Review by amandanguyeen — APPROVED** [review]

**Pull request merged**